lib/autoyast.pm

```diff
@@ -207,125 +207,287 @@ sub init_autoyast_profile {
-    my $errors      = $args{errors};
+    is_processable($node);
+
+  A node is considered 'processable' when:
+   - is a simple key-value pair where value is text:
+     quotas: 'true'
+   - has properties:
+     quotas:
+       _t: boolean
+       __text: 'true'
+  In other words, a node is not processable when just needs to be traversed in the YAML.
+
+=cut
+sub is_processable {
+    my $node = shift;
+    return (!ref $node ||
+          (ref $node eq 'HASH' && has_properties($node)));
+}
+
+=head2 has_properties
+
+  has_properties($node);
+
+- A XML node can be specified in YAML along with its attributes from XML:
+    XML:  <subvolumes t="list">
+    YAML: subvolumes:
+            _t: list
+    XPATH: ns:subvolumes[@t='list']
+
+- XPATH functions can be used:
+
+    'count': when you want to explicitly count the element in a list
+    XML:  <subvolumes t="list">
+            <subvolume t="map">
+    YAML: subvolumes:
+            __count: 8
+            subvolume:
+                ...
+    XPATH: ns:subvolumes[count(ns:subvolume)=8]
+
+    'text': when you want to check some attribute and the text itself
+    XML:  <quotas t="boolean">true</quotas>
+    YAML: quotas:
+            _t: boolean
+            __text: true
+    XPATH: ns:quotas[text()='true' and @t='boolean']
+
+    'descendant': by default the algorithm will try to identify an element
+    in a list using direct children, but with this property is it possible to
+    add children which are descendant but not necessarily direct children.
+    It is useful when XML structure are almost exactly the same for different
+    list items and the only difference is found in more deeper descendants,
+    therefore avoiding to return multiple result for an XPATH expression.
+
+    YAML:
+            drive:
+            - label: 
+                _descendant: any
+                __text: root_multi_btrfs
+                disklabel: none
+                partitions:
+                partition:
+                - filesystem: btrfs
+                    label: root_multi_btrfs
+            - label: 
+                _descendant: any
+                __text: test_multi_btrfs
+                disklabel: none        
+                partitions:
+                partition:
+                - filesystem: btrfs
+                    label: test_multi_btrfs
+
+=cut
+sub has_properties {
+    my $node = shift;
+    return 0 unless ref $node;
+    return scalar(grep {
+            $_ eq '_t'        ||
+              $_ eq '__text'  ||
+              $_ eq '__count' ||
+              $_ eq '_descendant'
+    } keys %{$node});
+}
+
+=head2 create_xpath_predicate
+
+    create_xpath_predicate($node);
+
+Based on the properties of the node will create a predicate for the XPATH expression.  
+
+=cut
+sub create_xpath_predicate {
+    my $node       = shift;
+    my @predicates = ();
+
+    if (has_properties($node)) {
+        push @predicates, "text()='$node->{__text}'"      if $node->{__text};
```

Comment:
Did you try cases with `<![CDATA[]]>`? I wonder how this library behaves in such case. AFAIR it should be ignored normally.

Reply:
I didn't, will require further implementation.

Reply:
Hmm, are you sure, because normally xpath should parse CDATA as text. Would be nice if we can be sure if it works or we need to address it in future.

Reply:
I know what you meant, yes it should be text in the cloned profile, it should work, I think I cover this in similar way that was before:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/autoyast.pm#L274 (when there is internal text node or not) We don't have an example to quickly check though.